src/core/connection.c

```diff
@@ -5369,6 +5630,60 @@ QuicConnParamSet(
         Status = QUIC_STATUS_SUCCESS;
         break;
 
+    case QUIC_PARAM_CONN_SERVER_ENABLE_RESUME_ZERORTT: {
+        if (BufferLength != sizeof(uint8_t) ||
+            *(uint8_t*)Buffer > QUIC_SERVER_RESUME_AND_ZERORTT) {
+                Status = QUIC_STATUS_INVALID_PARAMETER;
+                break;
+        }
+
+        if (Connection->State.Started) {
+            Status = QUIC_STATUS_INVALID_STATE;
+            break;
+        }
+
+        if (*(uint8_t*)Buffer > QUIC_SERVER_NO_RESUME &&
+            Connection->HandshakeTP == NULL && Connection->ResumedTP == NULL) {
+            //
+            // TODO: specific logic to enable/disable 0-RTT
+            //
+            uint8_t CurProcIndex = QuicLibraryGetCurrentPartition(); // N.B. this could result in the wrong pool being used on free()
+
+            Connection->HandshakeTP =
+                QuicPoolAlloc(&MsQuicLib.PerProc[CurProcIndex].TransportParamPool);
+            if (Connection->HandshakeTP == NULL) {
+                QuicTraceEvent(
+                    AllocFailure,
+                    "Allocation of '%s' failed. (%llu bytes)",
+                    "handshake TP",
+                    sizeof(QUIC_TRANSPORT_PARAMETERS));
+                Status = QUIC_STATUS_OUT_OF_MEMORY;
+                break;
+            }
+            QuicZeroMemory(Connection->HandshakeTP, sizeof(QUIC_TRANSPORT_PARAMETERS));
+
+            Connection->ResumedTP =
```

Comment:
Are the ResumedTP and HandshakeTP always allocated in the same scenarios? If so, why not allocate in a single block?

Reply:
I thought about that optimization once I got this working. However, after our discussion offline, I'm moving to a different design that only needs one, and not both.